UI/src/model/selectedLaneModelling.js

```diff
@@ -227,12 +227,25 @@
       });
     }
 
+    function getSideCodesForLinks() {
+      var selectedMainLane = _.find(self.selection, function (lane) {
+        return getLaneCodeValue(lane) == 1;
+      });
+      var mainLaneGroup = collection.getGroup(selectedMainLane);
+      var sideCodesMapped = mainLaneGroup.map(function(lane){
+        return {linkId: lane.linkId, sideCode: lane.sideCode};
+      });
+
+      return sideCodesMapped;
+    }
+
     self.save = function(isAddByRoadAddressActive) {
       eventbus.trigger(self.singleElementEvent('saving'));
       omitIrrelevantExpiredLanes();
 
       var linkIds = _.head(self.selection).linkIds;
       var sideCode = _.head(self.selection).sideCode;
+      var sideCodesForLinks = getSideCodesForLinks();
```

Comment:
onkos nämä valitujen kaistojen sideCode vai valitujen kaistojen pääkaistojen sideCode

Reply:
NewLane on myös joku sideCode. Miksi sitä ei voi käyttää?

Reply:
Siinä on luotavien kaistojen sideCodet aina tietylle linkille. Luodessa monta kaistaa kerralla käyttöliittymästä ei ole mahdollista luoda kummankin sideCoden kaistoja samalle linkille. Tuo tieto tuodaan siksi, että aikaisemmin kaistat luotiin vain tietyille linkeille tietyllä kaksinumeroisella kaistakoodilla ja muuta tietoa ei tarvittu. sideCodesForLinks kertoo nyt jokaisen luotavan kaistan sideCoden.

Reply:
Eikö toi sama tieto ole NewLane objectissa?

Reply:
Kaistahan menee yksi yhteen linkin kanssa kun me insert one to one  relaatio tyyppisesti joten tää alkaa vaikuttaa olevan turhaa työtä

Reply:
lanes: lanes.concat(omitUnrelevantProperties(assetsToBeExpired)) toi kohta

Reply:
Frontin puolella "luodaan" yksi mallikaista joka viedään JSON sanomassa backendille, tämän mallikaistan pohjalta luodaan linkIds perusteella tietyille linkeille kaistat backendin puolella, Oli varmaan aikaisemmin ok tapa kun oli kaksinumeroinen kaistakoodi tiedossa.  Voitaisiin muuttaa, että JSON sisältäisi jokaisen luotavan kaista olion tiedot, mutta tämä oma ratkaisu jossa tuodaan sideCodet erikseen kaikille linkeille tuntui helpommalta tavalta mukautua vanhaan tapaan. Selkemäpi tapa voisi kyllä olla, että viedään jokaisen luotavan kaistan tiedot JSON

Reply:
hyym ehkä voishan sen muuttaa niin että käytetään NewLane kokonaan kun siellä sellainen field jo on olemassa

Reply:
Joo, pitää vähän tutkia noita muita operaatioita joita tehdään backendin puolella, että pitääkö niitä miten muuttaa jos alkaa sanomaa muuttamaan

Reply:
se backEndin newLanes sisältää siis käytännössä vain mallit niistä luotavista kaistoista, eli jos luodaan vaikka viidelle linkille kaistat 2 ja 3 niin silloin newLanes sisältää mallit kaistalle 2 ja kaistalle 3. Eli sideCoden ottaminen tuolta newLane oliosta ei ilman isompia muutoksia onnistu, pitäisi koko logiikkaa muuttaa

Reply:
okey varmaan tällä sitten mennään, varmaan vaatis jotakin laajempaa remontia x aikana